src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -391,4 +392,24 @@ class BehandlerService(
             connection.commit()
         }
     }
+
+    private fun List<PartnerinfoResponse>.selectActiveBehandlerKontor(): PartnerinfoResponse? =
+        if (this.isEmpty())
+            null
+        else if (this.size == 1)
+            this.first()
+        else {
+            // If duplicate partnerId for a herId, prefer partnerId of kontor with latest
+            // timestamp for dialogmeldingEnabled, else pick largest partnerId
+            val pBehandlerKontor = this.mapNotNull { partnerInfoResponse ->
+                database.connection.use { it.getBehandlerKontor(PartnerId(partnerInfoResponse.partnerId)) }
+            }.filter {
+                it.dialogmeldingEnabled != null
+            }.maxByOrNull {
+                it.dialogmeldingEnabled!!
+            }
+            if (pBehandlerKontor != null) {
+                this.first { it.partnerId == pBehandlerKontor.partnerId.toInt() }
+            } else null
+        } ?: this.maxByOrNull { it.partnerId }
```

Comment:
Har `partnerId`-tallet faktisk en iboende "rekkefølge/størrelse" i seg, eller velger vi den største bare for å ikke velge random?

Reply:
Ja, det er vel mest for å ha en slags regel for å velge (men det kan godt være at det er det nyeste som har størst partnerId).